cropduster/static/cropduster/js/cropduster.js

```diff
@@ -365,20 +365,24 @@ window.CropDuster = {};
             var $thumb = $input.closest('.cropduster-form').find('.cropduster-images');
             $thumb.find('a').remove();
 
+            preview_img = {}
             $.each(data.sizes, function(i, size) {
                 if (!size || !size.name) return;
+
                 if (thumbData[size.name]) {
+                    if ($.isEmptyObject(preview_img)) {
+                        preview_img = {
+                            // use width and height of first suitable thumb for admin preview
+                            'image_url': data.previewUrl,
```

Comment:
Shouldn't this be `thumbData[size.name].url`?

Reply:
Huh, I suppose it could 👍  Worth removing the injection of the preview url into context or are we okay with leaving that for some possible future use cases?

Reply:
Oh, is this intended to correct an issue where the width and height don't match the preview image dimensions? If that's the case, we can return them from the crop view in the data, since they get passed in the POST data on the form submission. Look at the `upload` view to see how to retrieve it.

Reply:
Otherwise, wouldn't the image be a 404 if `CROPDUSTER_CREATE_THUMBS` is false?

Reply:
Oh, true dat 👍  Less so to correct that (although..good point), so much as to correct for missing heights from the sizes object the preview image had been sourcing the width/height from before. In any case, will update to pass the preview height/width into context.

Reply:
Argh, i'm a dummy 🤦  I can't figure out a good way of injecting the preview dimensions into the context in all the places (short of opening the image file). And since we don't save preview thumbs to the database, I'm not sure how we would actually do so [here](https://github.com/theatlantic/django-cropduster/blob/master/cropduster/forms.py#L31), in particular. Which brings me back to the initial choice to use the `previewUrl` + the dimensions of one of the `thumbData` thumbs. It at least renders _an_ image in close enough of an aspect ratio not to raise too many objections, I think? 🤷

Reply:
I might have overlooked something obvious: isn't the preview width and height determined by a setting? Specifically, `cropduster.settings` has `CROPDUSTER_PREVIEW_WIDTH` and `CROPDUSTER_PREVIEW_HEIGHT`?

I think this could be accomplished in a manner analogous to how you passed the preview url (as widget context data + data attributes on the input, and the JSON response of the crop view), only pulling the value from those settings.

Reply:
You'd think so but it seems more dynamic than that- see e.g. [these](https://cdn.theatlantic.com/assets/media/img/mt/2020/07/GettyImages_50568456/_preview.jpg) [example](https://cdn.theatlantic.com/assets/media/img/mt/2020/07/h_15352474_toned-1/_preview.jpg) [images](https://cdn.theatlantic.com/assets/media/img/mt/2020/07/2/_preview.jpg). I think because of this [bit](https://github.com/theatlantic/django-cropduster/blob/master/cropduster/views/__init__.py#L92-L107) in the views and the other places where it looks like the `CROPDUSTER_PREVIEW_W/H` serve more as defaults than strict dimensions to use.

Reply:
That bit in the views is for standalone images, which have a preview size distinct from the usual cropduster preview. We can safely ignore that case because this is for showing the preview on the admin inline, which standalone images don't have.

But you're right that the setting isn't an exact measure. It's actually sets the maximum width and height of the preview image. Either the width or the height might be smaller, depending on the aspect ratio of the uploaded image.

https://github.com/theatlantic/django-cropduster/blob/11a480c63ab923b306b8c13f6c2fb95e50edd805/cropduster/views/__init__.py#L249-L263

So you should still be able to determine the preview width and height given the original image size and the value in those settings.

Reply:
Ah, cheers. Okay, I'll give that a try 👍